services/blockchain-connector/shared/sdk/network.js

```diff
@@ -39,6 +39,11 @@ const refreshNetworkStatus = async () => {
 			logger.warn(`Error occurred while refreshing network status:\n${err.stack}`);
 		}
 	};
+
+	if (!networkStatus) {
+		await refreshNetworkStatusListener();
+	}
```

Comment:
I think this part is not correct, We need to update network status on every new block even if already exists in-memory. There are some properties which is getting updated on every new block, for e.g. `finalizedHeight`
```suggestion
	if (!networkStatus) {
		await refreshNetworkStatusListener();
	}
```

Reply:
This is executed only once at init. Required, in case the cache isn't set before the event listener kicks in and the indexer is querying for the network status.

Reply:
Thanks for the clarification 👍